From d0a13a735f7522c43b5d5a4015974ef113b6beee Mon Sep 17 00:00:00 2001
From: Sujeet Baranwal <sbaranwal@marvell.com>
Date: Tue, 30 Apr 2019 17:06:04 -0700
Subject: [PATCH 204/386] mmc: Correction in mmc slot switching

Switching from one slot to another doesn't need bus id to be
programmed to zero. This step was added in past mistakenly
which is now removed as part of the solution.

Change-Id: Ifd80c5f62b31fea3a4f71fc892dac20deb2aeb0a
Signed-off-by: Sujeet Baranwal <sbaranwal@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/8144
Reviewed-by: Sujeet Kumar Baranwal <Sujeet.Baranwal@cavium.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/mmc/host/cavium.c | 10 +++-------
 drivers/mmc/host/cavium.h |  8 --------
 2 files changed, 3 insertions(+), 15 deletions(-)

diff --git a/drivers/mmc/host/cavium.c b/drivers/mmc/host/cavium.c
index c43691478623..6f68eb63dc80 100644
--- a/drivers/mmc/host/cavium.c
+++ b/drivers/mmc/host/cavium.c
@@ -328,7 +328,7 @@ static void do_switch(struct cvm_mmc_host *host, u64 emm_switch)
 	 * Modes setting only taken from slot 0. Work around that hardware
 	 * issue by first switching to slot 0.
 	 */
-	if (bus_id && errata_29956(host)) {
+	if (bus_id) {
 		clear_bus_id(&emm_switch);
 		writeq(emm_switch, host->base + MIO_EMM_SWITCH(host));
 		set_bus_id(&emm_switch, bus_id);
@@ -432,12 +432,8 @@ static void cvm_mmc_switch_to(struct cvm_mmc_slot *slot)
 
 	writeq(slot->cached_rca, host->base + MIO_EMM_RCA(host));
 	emm_switch = slot->cached_switch;
-	/* Due to errata 29956 the clock is not set properly when the
-	 * bus_id is non-zero.
-	 */
-	set_bus_id(&emm_switch, 0);
-	do_switch(host, emm_switch & MIO_EMM_SWITCH_CLK);
-	set_bus_id(&emm_switch, slot->bus_id);
+
+
 	do_switch(host, emm_switch);
 	host->powered = true;
 
diff --git a/drivers/mmc/host/cavium.h b/drivers/mmc/host/cavium.h
index 5b2d67b3436e..0b6196a5bb11 100644
--- a/drivers/mmc/host/cavium.h
+++ b/drivers/mmc/host/cavium.h
@@ -314,12 +314,4 @@ static inline bool is_mmc_otx2_A0(struct cvm_mmc_host *host)
 	return (pdev->revision == 0x00) &&
 		(chip_id == PCI_SUBSYS_DEVID_9XXX);
 }
-
-static inline bool errata_29956(struct cvm_mmc_host *host)
-{
-	/* Clock is not set properly when the
-	 * bus_id is non-zero.
-	 */
-	return is_mmc_otx2_A0(host);
-}
 #endif
-- 
2.17.1

